tests/dataset/tiling/test_lvae_tiled_patching.py

```diff
@@ -0,0 +1,188 @@
+import numpy as np
+import pytest
+
+from careamics.config.tile_information import TileInformation
+from careamics.dataset.tiling.lvae_tiled_patching import (
+    compute_padding,
+    compute_tile_grid_shape,
+    compute_tile_info,
+    extract_tiles,
+    n_tiles_1d,
+    total_n_tiles,
+)
+from careamics.prediction_utils.stitch_prediction import stitch_prediction
+
+
+@pytest.mark.parametrize(
+    "data_shape, tile_size, overlaps",
+    [
+        # 2D
+        ((1, 3, 10, 9), (4, 4), (2, 2)),
+        ((1, 3, 10, 9), (8, 8), (4, 4)),
+        # 3D
+        ((1, 3, 8, 16, 17), (4, 4, 4), (2, 2, 2)),
+        ((1, 3, 8, 16, 17), (8, 8, 8), (4, 4, 4)),
+    ],
+)
+def test_extract_tiles(data_shape, tile_size, overlaps):
+    """Test extracted tiles are all the same size and can reconstruct the image."""
+
+    arr = np.random.random_sample(data_shape).astype(np.float32)
+
+    tile_data_generator = extract_tiles(
+        arr=arr, tile_size=np.array(tile_size), overlaps=np.array(overlaps)
+    )
+
+    tiles = []
+    tile_infos = []
+
+    # Assemble all tiles and their respective coordinates
+    for tile, tile_info in tile_data_generator:
+
+        overlap_crop_coords = tile_info.overlap_crop_coords
+        stitch_coords = tile_info.stitch_coords
+
+        # add data to lists
+        tiles.append(tile)
+        tile_infos.append(tile_info)
+
+        # check tile shape, ignore channel dimension
+        assert tile.shape[1:] == tile_size
+        assert len(overlap_crop_coords) == len(stitch_coords) == len(tile_size)
+
+    # stitch_prediction returns list
+    stitched_arr = stitch_prediction(tiles, tile_infos)[0]
+
+    np.testing.assert_array_equal(arr, stitched_arr)
```

Comment:
I see you already have both integration and unit tests in place.  If they are passing, then all is good !!